packages/@react-stately/toast/src/useToastState.ts

```diff
@@ -10,57 +10,233 @@
  * governing permissions and limitations under the License.
  */
 
-import {ReactNode, useRef, useState} from 'react';
-import {Timer} from './';
-import {ToastProps, ToastState, ToastStateValue} from '@react-types/toast';
+import {useCallback, useMemo} from 'react';
+import {useSyncExternalStore} from 'use-sync-external-store/shim';
 
-interface ToastStateProps {
-  value?: ToastStateValue[]
+export interface ToastStateProps {
+  /** The maximum number of toasts to display at a time. */
+  maxVisibleToasts?: number,
+  /**
+   * Whether toasts have an exit animation. If true, toasts are not
+   * removed immediately but transition into an "exiting" state instead.
+   * Once the animation is complete, call the `remove` function.
+   */
+  hasExitAnimation?: boolean
 }
 
-const TOAST_TIMEOUT = 5000;
+export interface ToastOptions {
+  /** Handler that is called when the toast is closed, either by the user or after a timeout. */
+  onClose?: () => void,
+  /** A timeout to automatically close the toast after, in milliseconds. */
+  timeout?: number,
+  /** The priority of the toast relative to other toasts. */
```

Comment:
Nit: maybe add to this stating that a higher number indicates a higher priority just for clarity.

Reply:
```suggestion
  /** The priority of the toast relative to other toasts. Larger numbers indicate higher priority. */
```